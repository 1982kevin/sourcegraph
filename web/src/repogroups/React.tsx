import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
