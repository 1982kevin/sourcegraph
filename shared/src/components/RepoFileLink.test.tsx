import React from 'react'
import { RepoFileLink } from './RepoFileLink'
import { mount } from 'enzyme'

describe('RepoFileLink', () => {
    test('renders', () => {
        const component = mount(
            <RepoFileLink
                repoName="example.com/my/repo"
                repoURL="https://example.com"
                filePath="my/file"
                fileURL="https://example.com/file"
            />
        )
        expect(component).toMatchSnapshot()
    })
})
