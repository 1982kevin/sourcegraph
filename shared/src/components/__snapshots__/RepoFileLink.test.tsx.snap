// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoFileLink renders 1`] = `
Array [
  <AnchorLink
    to="https://example.com"
  >
    <a
      href="https://example.com"
    >
      my/repo
    </a>
  </AnchorLink>,
  " â€º",
  " ",
  <AnchorLink
    to="/file?subtree=true"
  >
    <a
      href="/file?subtree=true"
    >
      my/
      <strong>
        file
      </strong>
    </a>
  </AnchorLink>,
]
`;
